adaptive_scheduler/models.py

```diff
@@ -10,35 +10,48 @@
 
 # Required for true (non-integer) division
 from __future__ import division
+from pickletools import optimize
 
+from time_intervals.intervals import Intervals
 from rise_set.sky_coordinates import RightAscension, Declination
 from rise_set.astrometry import (make_ra_dec_target,
                                  make_minor_planet_target,
                                  make_major_planet_target,
                                  make_comet_target,
-                                 make_satellite_target)
+                                 make_satellite_target,
+                                 calculate_airmass_at_times)
 from rise_set.angle import Angle
 from rise_set.exceptions import InvalidAngleError, AngleConfigError, RatesConfigError
 from rise_set.rates import ProperMotion
-from adaptive_scheduler.utils import (iso_string_to_datetime, convert_proper_motion,
-                                      EqualityMixin, safe_unidecode)
+from adaptive_scheduler.utils import (iso_string_to_datetime, convert_proper_motion, datetime_to_normalised_epoch,
+                                      EqualityMixin, safe_unidecode, normalise_datetime_intervals)
 from adaptive_scheduler.printing import plural_str as pl
-from adaptive_scheduler.kernel.reservation_v3 import CompoundReservation_v2 as CompoundReservation
+from adaptive_scheduler.kernel.reservation import CompoundReservation
 from adaptive_scheduler.feedback import UserFeedbackLogger
 from adaptive_scheduler.eventbus import get_eventbus
 from adaptive_scheduler.moving_object_utils import required_fields_from_scheme
 from adaptive_scheduler.observation_portal_connections import ObservationPortalConnectionError
 
-from datetime import datetime
+from datetime import datetime, timedelta
 from collections import defaultdict
+import numpy as np
 import ast
+import os
+import json
 import logging
 import random
+from redis import Redis
 
 log = logging.getLogger(__name__)
 
 event_bus = get_eventbus()
 
+redis_instance = Redis.from_url(url=os.getenv('REDIS_URL', 'redis://redis'), socket_connect_timeout=15,
+              socket_timeout=30)
+
+
+AIRMASS_WEIGHTING_COEFFICIENT = 0.1
```

Comment:
This is something we may want to make settable later... for now it is matching the weighting of the earliest windows (i.e. earliest window or best airmass should both add 0.1 to the priority, which is miniscule compared to the potentially hundreds of thousands the priority could be).